File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int y = Integer.parseInt(st.nextToken());
        int x = Integer.parseInt(st.nextToken());

        char[][] array = new char[y][x];

        for (int i = 0; i < y; i++) {
            array[i] = br.readLine().toCharArray();
        }

        int answer = 64;

        for (int i = 0; i <= y - 8; i++) {
            for (int j = 0; j <= x - 8; j++) {
                char firstChar = array[i][j];
                int count = 0;

                for (int k = 0; k < 8; k++) {
                    for (int l = 0; l < 8; l++) {
                        if (k % 2 == 0) {
                            if (l % 2 == 0 && array[i + k][j + l] != firstChar) {
                                count++;
                            } else if (l % 2 != 0 && array[i + k][j + l] == firstChar) {
                                count++;
                            }
                        } else {
                            if (l % 2 == 0 && array[i + k][j + l] == firstChar) {
                                count++;
                            } else if (l % 2 != 0 && array[i + k][j + l] != firstChar) {
                                count++;
                            }
                        }
                    }
                }

                count = Math.min(count, 64 - count);
                answer = Math.min(answer, count);
            }
        }

        System.out.println(answer);
    }
}

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.kt
fun main() {
    var answer = 64
    val (y, x) = readln().split(" ").map { it.toInt() }
    val array = Array(y) { charArrayOf() }

    for (i in 0 until y) {
        val input = readln().toCharArray()
        array[i] = input
    }

    for (i in 0..y - 8) {
        for (j in 0..x - 8) {
            val firstChar = array[i][j]
            var count = 0

            for (k in 0..7) {
                for (l in 0..7) {
                    when {
                        k % 2 == 0 -> {
                            if (l % 2 == 0 && array[i + k][j + l] != firstChar) count++
                            else if (l % 2 != 0 && array[i + k][j + l] == firstChar) count++
                        }
                        else -> {
                            if (l % 2 == 0 && array[i + k][j + l] == firstChar) count++
                            else if (l % 2 != 0 && array[i + k][j + l] != firstChar) count++
                        }
                    }
                }
            }

            count = minOf(count, 64 - count)
            if (answer > count) answer = count
        }
    }

    println(answer)
}